src/AcceptanceTesting/SpecByExample/SFA.DAS.Payments.AcceptanceTests/TableParsers/CommitmentsTableParser.cs

```diff
@@ -194,6 +197,12 @@ private static CommitmentReferenceData ParseCommitmentsTableRow(TableRow row, Co
                 throw new ArgumentException($"'{row[structure.EffectiveToIndex]}' is not a valid effective to");
             }
 
+            DateTime? transferApprovalDate = null;
+            if (structure.TransferApprovalDateIndex > -1 && !TryParseNullableDateTime(row[structure.TransferApprovalDateIndex], out transferApprovalDate))
+            {
+                throw new ArgumentException($"'{row[structure.TransferApprovalDateIndex]}' is not a valid effective from");
+            }
+
             var standardCode = row.ReadRowColumnValue<long>(structure.StandardCodeIndex, "standard code", Defaults.StandardCode);
```

Comment:
generic <long> not needed here.

Reply:
I know resharper says it's not needed, but it's unclear what type is returned unless you look at the implementation of the class. Would rather keep it here

Reply:
I don't see a generic in the changed code?